runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/ThreeStepCrypticClue.java

```diff
@@ -80,7 +94,24 @@ public static ThreeStepCrypticClue forText(String plainText, String text)
 		return new ThreeStepCrypticClue(steps, plainText);
 	}
 
-	private final List<Map.Entry<CrypticClue, Boolean>> clueSteps;
+	@Nullable
+	@VisibleForTesting
+	static CrypticClue getCrypticFromText(final String rawText)
+	{
+		for (CrypticClue clue : CrypticClue.CLUES)
+		{
+			if (!rawText.equalsIgnoreCase(clue.getText()))
+			{
+				continue;
+			}
+
+			return clue;
+		}
+		return null;
+	}
+
+	@VisibleForTesting
+	final List<Map.Entry<CrypticClue, Boolean>> clueSteps;
```

Comment:
SortedMap?

Reply:
We don't want to sort by keys though, we want to maintain insertion order.

Reply:
I opted to change this to a `List<ThreeStepCrypticClueStep>`, with that class being a simple POJO:

```java
@AllArgsConstructor
@Data
class ThreeStepCrypticClueStep
{
	final CrypticClue clue;
	boolean completed;
}
```